pkg/app/piped/platformprovider/ecs/client.go

```diff
@@ -472,17 +472,19 @@ func (c *client) getLoadBalancerArn(ctx context.Context, targetGroupArn string)
 	return output.TargetGroups[0].LoadBalancerArns[0], nil
 }
 
-func (c *client) ModifyListeners(ctx context.Context, listenerArns []string, routingTrafficCfg RoutingTrafficConfig) error {
+func (c *client) ModifyListeners(ctx context.Context, listenerArns []string, routingTrafficCfg RoutingTrafficConfig) ([]string, error) {
 	if len(routingTrafficCfg) != 2 {
-		return fmt.Errorf("invalid listener configuration: requires 2 target groups")
+		return nil, fmt.Errorf("invalid listener configuration: requires 2 target groups")
 	}
 
+	modifiedRuleArns := make([]string, 0)
+
 	for _, listenerArn := range listenerArns {
 		describeRulesOutput, err := c.elbClient.DescribeRules(ctx, &elasticloadbalancingv2.DescribeRulesInput{
 			ListenerArn: aws.String(listenerArn),
 		})
 		if err != nil {
-			return fmt.Errorf("failed to describe rules of listener %s: %w", listenerArn, err)
+			return nil, fmt.Errorf("failed to describe rules of listener %s: %w", listenerArn, err)
```

Comment:
Sorry I missed this.
IMO, we should return `modifiedRuleArns` here because we must return modified arns when an error occurs on non-first listenerArn.
```suggestion
			return modifiedRuleArns, fmt.Errorf("failed to describe rules of listener %s: %w", listenerArn, err)
```

Reply:
I agree